geode-core/src/main/java/org/apache/geode/cache/control/RestoreRedundancyResults.java

```diff
@@ -0,0 +1,148 @@
+   */
+  Status getStatus();
+
+  /**
+   * Returns a message describing the results of this restore redundancy operation.
+   *
+   * @return A {@link String} describing the results of this restore redundancy operation.
+   */
+  String getMessage();
+
+  /**
+   * Returns the {@link RegionRedundancyStatus} for a specific region or null if that region
+   * is not present in this {@link RestoreRedundancyResults}.
+   *
+   * @param regionName The region to which the {@link RegionRedundancyStatus} to be returned
+   *        belongs.
+   * @return A {@link RegionRedundancyStatus} for the specified region or null if that region is not
+   *         present in this {@link RestoreRedundancyResults}.
+   */
+  RegionRedundancyStatus getRegionResult(String regionName);
+
+  /**
+   * Returns all the {@link RegionRedundancyStatus RegionRedundancyStatuses} for regions with
+   * configured redundancy but zero actual redundant copies.
+   *
+   * @return A {@link Map} of {@link String} region name to {@link RegionRedundancyStatus} for every
+   *         region contained in this {@link RestoreRedundancyResults} with configured redundancy
+   *         but zero actual redundant copies.
+   */
+  Map<String, RegionRedundancyStatus> getZeroRedundancyRegionResults();
+
+  /**
+   * Returns all the {@link RegionRedundancyStatus RegionRedundancyStatuses} for regions with with
+   * at least one redundant copy, but fewer than the configured number of redundant copies.
+   *
+   * @return A {@link Map} of {@link String} region name to {@link RegionRedundancyStatus} for every
+   *         region contained in this {@link RestoreRedundancyResults} with at least one redundant
+   *         copy, but fewer than the configured number of redundant copies.
+   */
+  Map<String, RegionRedundancyStatus> getUnderRedundancyRegionResults();
+
+  /**
+   * Returns all the {@link RegionRedundancyStatus RegionRedundancyStatuses} for regions with
+   * redundancy satisfied.
+   *
+   * @return A {@link Map} of {@link String} region name to {@link RegionRedundancyStatus} for every
+   *         region contained in this {@link RestoreRedundancyResults} with redundancy satisfied.
+   */
+  Map<String, RegionRedundancyStatus> getSatisfiedRedundancyRegionResults();
+
+  /**
+   * Returns all the {@link RegionRedundancyStatus RegionRedundancyStatuses} contained in this
+   * {@link RestoreRedundancyResults}. This method may return the actual backing map depending on
+   * implementation.
+   *
+   * @return A {@link Map} of {@link String} region name to {@link RegionRedundancyStatus} for every
+   *         region contained in this {@link RestoreRedundancyResults}.
+   */
+  Map<String, RegionRedundancyStatus> getRegionResults();
+
+  /**
+   * Returns the total number of primaries that were transferred as part of the restore redundancy
+   * operations.
+   *
+   * @return the total number of primaries that were transferred
+   */
+  int getTotalPrimaryTransfersCompleted();
```

Comment:
Is it reasonable to assume we will never be able to transfer 2^31 or more primaries?

Reply:
I would hope we don't have to transfer that many primaries... ever...Can we even handle that many buckets (not theoretically but do we know of any cases where we have over 100 million buckets❓)

Reply:
The most primaries that could possibly be transferred for a single region during a restore redundancy operation is equal to the number of buckets in the region (this scenario would also require there to be at least as many servers as there are buckets). Assuming the default number of buckets (113) is used, there would need to be a little over 19 million regions for an integer overflow to occur, which seems like an extremely unlikely scenario.